src/ui/utils.ts

```diff
@@ -1,7 +1,134 @@
+            scrollbar-color: ${Color} ${BackgroundColor} !important;
+        }
+
+        .color-picker--focused .color-picker__wrapper {
+            border: unset !important;
+            outline: 2px solid ${BackgroundColor};
+        }
+
+        .check-button .checkbox__input ~ .checkbox__checkmark {
+            background-color: ${ActiveBackgroundColor} !important;
+        }
+
+        .preview .settings-button-icon, .preview .m-help-button__text::before, .reset-button__icon, .dropdown__selected::after, .preview .theme-group__more-button::after, .preview .collapsible-panel__group__expand-button::before, .preview .page__back-button::before, .nav-button::after, .preview .m-help-button::after, .check-button .checkbox__input ~ .checkbox__checkmark {
+            filter: invert(${Light});
+        }
+
+        .color-picker__input, .color-picker__wrapper {
+            background-color: unset !important;
+        }
+
+        .site-list__item__remove:hover {
+            color: ${Theme.colors.popup} !important;
+        }
+
+        .preview .theme-group__presets-wrapper::before {
+            border-color: ${BackgroundColor} !important;
+        }
+
+        .loader__message {
+            display: none;
+        }
+
+        .textbox::placeholder {
+            color: ${Color};
+        }
+
+        .header__more-settings__shortcut-wrapper--set svg {
+            --icon-color: ${Color};
+        }
+
+        .tab-panel__buttons::before, .tab-panel__buttons::after {
+            border-bottom-color: ${BackgroundColor};
+        }
+
+        .footer-help-link:hover, .donate-link:hover, .site-list__textbox:focus, .automation__behavior .dropdown, .preview .theme-preset-picker .dropdown__list {
+            box-shadow: unset;
+        }
+
+        .checkbox__checkmark::before, .checkbox__checkmark::after, .updown__icon::before, .updown__icon::after, .checkbox__input:checked + .checkbox__checkmark::before, .checkbox__input:checked + .checkbox__checkmark::after, .select__expand__icon::before, .select__expand__icon::after, .check-button .checkbox__checkmark, .slider__thumb, .site-list__item__remove:hover {
+            background-color: ${Color};
+        }
+
+        .header__more-settings::after, .news::before, .news__header {
+            background-image: unset;
+        }
+
+        html, button, .tab-panel__tab, .toggle__btn, .updown__icon, .track--clickable, .toggle::before, .header__more-settings, .multi-switch, .checkbox, .header__more-settings__shortcut, .news__list, .select__option, .select__list, .news__header, .check-button .checkbox__checkmark, .dropdown__list__item, .message-box, .dropdown__list, .editor, .preview .page, .preview .news-section, .preview .news-section__popover, .color-picker__hsb-line, .site-list__textbox, .time-range-picker__input, .automation__location-control__latitude, .automation__location-control__longitude, .text-list__textbox, .select__textbox, .dynamic-per-site__search-input {
```

Comment:
Listing selectors within the app is absolutely not scaleable and unmaintainable. CSS variables (custom properties) should be used and redeclared like `:root { --color-bg: #000; ... }`.

Reply:
Do you mean the entire css of the entire extension should be rewritten for this?

Reply:
We have `theme.less` file with LESS variables. In theory you should be able to have something like:
```less
:root {
    --color-back: #141e24;
}

@color-back: var(--color-back);
```

And then you can insert CSS like
```less
:root {
    --color-back: #000000;
}
```
and the colors within the entire UI should change. However there could be issues with the colors of icons (they should be moved from LESS to HTML/JSX markup (`currentcolor` can be used for SVG to make it use CSS `color` value). Another issue could be LESS color mixing functions (like `fade` or `mix`), they won't process the CSS variables.

Please understand that even if you do all those changes, the PR most likely won't be merged due to reasons I described earlier. However it could be useful to replace LESS color variables with CSS variables in our code. If you would like to have some practice, you can submit a separate PR addressing these color variables. Then another for moving icons from CSS to SVG (we have  `src/ui/icons` folder, see how they are used in the code).

Reply:
@alexanderby I did clean it up as much as I could (while still using css selectors/classes). I'd like an answer to my question. If it wouldn't be merged even with using css variables, I'll just close the pr/issue and keep it for my own usage